sqlglot/transforms.py

```diff
@@ -55,6 +55,18 @@ def _to_sql(self, expression: exp.Expression) -> str:
     return _to_sql
 
 
+def unnest_generate_series(expression: exp.Expression) -> exp.Expression:
+    """Unnests GENERATE_SERIES or SEQUENCE table references."""
+    if isinstance(expression, exp.Table):
+        if isinstance(expression.this, exp.GenerateSeries):
+            unnest = exp.Unnest(expressions=[expression.this])
+
+            if expression.alias:
+                return exp.alias_(unnest, alias="_u", table=[expression.alias], copy=False)
+            return unnest
+    return expression
```

Comment:
Small nit, could we refactor this to make it a bit less convoluted? Maybe merging the two branches (or negating them to `return expression` early), caching the common expressions such as `expression.this` etc